azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -153,8 +179,16 @@ public int createImageType(final ImageType imageType) {
       imageTypes = this.databaseOperator
           .query(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_NAME, fetchImageTypeHandler,
               name.toLowerCase());
+      if (imageTypes == null) {
```

Comment:
A non-null result is not guaranteed?

Reply:
For the code, I think yes, but when we mock things with our tests, I think not.

Reply:
Shouldn't we fix the tests then?

Reply:
For example this conditional block here is not needed as it will never be executed, why not simplify and optimize the code?

Reply:
Thank you for making me double check this. I was wrong.